### PR TITLE
add descriptions to README

### PR DESCRIPTION
**Pull request by SaranjeetKaur**
Added brief descriptions of `metric`, `leapfrog integration`, `proposal for trajectories` in the README.

### CODE REVIEW

**Comment by SaranjeetKaur** [comment]
@xukai92, @yebai - please review the above descriptions in the README and let me know your comments.

**Review by xukai92 — COMMENTED** [review]
I really like the detailed content but I feel it may be a little bit too techinical to start with (as in the README). Maybe we could use the `details` tag (https://www.w3schools.com/tags/tag_details.asp) to make the math part expand when clicked?

**Comment by SaranjeetKaur** [comment]
@xukai92 - nice idea! I've made the changes, please have a look and let me know if it looks ok or any change is needed

File: README.md [hunk 1]
```diff
@@ -29,6 +29,26 @@ If you are interested in using AdvancedHMC.jl through a probabilistic programmin
 
 ## A minimal example - sampling from a multivariate Gaussian using NUTS
 
+In this section we demonstrate a minimal example of sampling from a multivariate Gaussian (10 dimensional) using the no U-turn sampler (NUTS). Below we describe the major components of the Hamiltonian system which are essential to sample using this approach:
+
+- **Metric**: In many sampling problems the sample space is usually associated with a metric, that allows us to measure the distance between any two points, and other similar quantities. In the example in this section, we use a special metric called the **Euclidean Metric**, represented with a `D × D` matrix from which we can compute distances. 
+<details>
+ <summary>Further details about the Metric component</summary>
+ The Euclidean metric is also known as the mass matrix in the physical perspective. For available metrics refer [Hamiltonian mass matrix](https://github.com/SaranjeetKaur/AdvancedHMC.jl#hamiltonian-mass-matrix-metric).
```
Comment (xukai92) [inline README.md:37]:
```suggestion
 The Euclidean metric is also known as the mass matrix in the physical perspective. For available metrics refer <a href="https://github.com/SaranjeetKaur/AdvancedHMC.jl#hamiltonian-mass-matrix-metric">Hamiltonian mass matrix</a>.
```

File: README.md [hunk 2]
```diff
@@ -29,6 +29,26 @@ If you are interested in using AdvancedHMC.jl through a probabilistic programmin
...
+<details>
+ <summary>About the leapfrog integration scheme</summary>
+ Suppose ${\bf x}$ and ${\bf v}$ are the position and velocity of an individual particle respectively; $i$ and $i+1$ are the indices for time values $t_i$ and $t_{i+1}$ respectively; $dt = t_{i+1} - t_i$ is the time step size (constant and regularly spaced intervals); and ${\bf a}$ is the acceleration induced on a particle by the forces of all other particles. Furthermore, suppose positions are defined at times $t_i, t_{i+1}, t_{i+2}, \dots $, spaced at constant intervals $dt$, the velocities are defined at halfway times in between, denoted by $t_{i-1/2}, t_{i+1/2}, t_{i+3/2}, \dots $, where $t_{i+1} - t_{i + 1/2} = t_{i + 1/2} - t_i = dt / 2$, and the accelerations ${\bf a}$ are defined only on integer times, just like the positions. Then the leapfrog integration scheme is given as: $x_{i} = x_{i-1} + v_{i-1/2} dt; \quad v_{i+1/2} = v_{i-1/2} + a_i dt$. For available integrators refer [Integrator](https://github.com/SaranjeetKaur/AdvancedHMC.jl#integrator-integrator).
```
Comment (xukai92) [inline README.md:43]:
```suggestion
 Suppose ${\bf x}$ and ${\bf v}$ are the position and velocity of an individual particle respectively; $i$ and $i+1$ are the indices for time values $t_i$ and $t_{i+1}$ respectively; $dt = t_{i+1} - t_i$ is the time step size (constant and regularly spaced intervals); and ${\bf a}$ is the acceleration induced on a particle by the forces of all other particles. Furthermore, suppose positions are defined at times $t_i, t_{i+1}, t_{i+2}, \dots $, spaced at constant intervals $dt$, the velocities are defined at halfway times in between, denoted by $t_{i-1/2}, t_{i+1/2}, t_{i+3/2}, \dots $, where $t_{i+1} - t_{i + 1/2} = t_{i + 1/2} - t_i = dt / 2$, and the accelerations ${\bf a}$ are defined only on integer times, just like the positions. Then the leapfrog integration scheme is given as: $x_{i} = x_{i-1} + v_{i-1/2} dt; \quad v_{i+1/2} = v_{i-1/2} + a_i dt$. For available integrators refer <a href="https://github.com/SaranjeetKaur/AdvancedHMC.jl#integrator-integrator">Integrator</a>.
```

File: README.md [hunk 3]
```diff
@@ -29,6 +29,26 @@ If you are interested in using AdvancedHMC.jl through a probabilistic programmin
...
+</details>
+
+- **Proposal for trajectories (static or dynamic)**: Different types of proposals can be used, which maybe static or dynamic. At each iteration of any variant of the HMC algorithm there are two main steps - the first step changes the momentum and the second step may change both the position and the momentum of a particle. 
+<details>
+ <summary>More about the proposals</summary>
+ In the classical HMC approach, during the first step, new values for the momentum variables are randomly drawn from their Gaussian distribution, independently of the current values of the position variables. Whereas, during the second step, a Metropolis update is performed, using Hamiltonian dynamics to provide a new state. For available proposals refer [Proposal](https://github.com/SaranjeetKaur/AdvancedHMC.jl#proposal-proposal).
```
Comment (xukai92) [inline README.md:49]:
```suggestion
 In the classical HMC approach, during the first step, new values for the momentum variables are randomly drawn from their Gaussian distribution, independently of the current values of the position variables. Whereas, during the second step, a Metropolis update is performed, using Hamiltonian dynamics to provide a new state. For available proposals refer <a href="https://github.com/SaranjeetKaur/AdvancedHMC.jl#proposal-proposal">Proposal</a>.
```

File: README.md (hunk 1 above)
Comment (SaranjeetKaur) [reply README.md:37]: good catch! thanks for that!

**Review by xukai92 — DISMISSED** [review]

File: README.md [hunk 4]
```diff
@@ -29,6 +29,26 @@ If you are interested in using AdvancedHMC.jl through a probabilistic programmin
 
 ## A minimal example - sampling from a multivariate Gaussian using NUTS
 
+In this section we demonstrate a minimal example of sampling from a multivariate Gaussian (10 dimensional) using the no U-turn sampler (NUTS). Below we describe the major components of the Hamiltonian system which are essential to sample using this approach:
+
+- **Metric**: In many sampling problems the sample space is usually associated with a metric, that allows us to measure the distance between any two points, and other similar quantities. In the example in this section, we use a special metric called the **Euclidean Metric**, represented with a `D × D` matrix from which we can compute distances. 
+<details>
+ <summary>Further details about the Metric component</summary>
+ The Euclidean metric is also known as the mass matrix in the physical perspective. For available metrics refer <a href="https://github.com/SaranjeetKaur/AdvancedHMC.jl#hamiltonian-mass-matrix-metric">Hamiltonian mass matrix</a>.
```
Comment (yebai) [inline README.md:37]: We need to avoid these hard-coded links: `https://github.com/SaranjeetKaur/AdvancedHMC.jl`. There are a few more in other places.

**Review by yebai — CHANGES_REQUESTED** [review]

**Comment by SaranjeetKaur** [comment]
@yebai - have updated it now

**Pull request merged**